src/Data/Array/Mutable/Linear.hs

```diff
@@ -60,31 +64,42 @@ import GHC.Exts hiding (toList, fromList)
 import GHC.Stack
 import qualified Unsafe.Linear as Unsafe
 import qualified Unsafe.MutableArray as Unsafe
-import Prelude.Linear ((&))
-import Prelude hiding (length, read)
+import Prelude.Linear ((&), forget)
+import qualified Control.Monad as Control
+import Prelude hiding (size, read)
 import qualified Prelude
 
 -- # Data types
 -------------------------------------------------------------------------------
 
 data Array a where
-  Array :: Int -> (MutableArray# RealWorld a) -> Array a
+  Array :: MutableArray# RealWorld a -> Array a
```

Comment:
Now the `Array` does not have an extra size parameter, since `MutableArray#` already comes with a `sizeof` method.